Automattic/wp-calypso
# Purchases: Perform paths cleanup in Manage Purchases section

This PR tries to refactor Manage Purchases section to use Calypso url and sections patterns.

### Changes introduced
1. For all url starting with `/purchases/*` add `/me` prefix, i.e. https://wordpress.com/me/purchases in Calypso.
2. Rename https://wordpress.com/payment-methods/add-credit-card to https://wordpress.com/me/purchases/add-credit-card in Calypso.
3. Rename https://wordpress.com/me/billing to https://wordpress.com/me/purchases/billing.
4. Merge section chunks related to purchases, billing and payment methods.
5. Add url redirects to make sure all old urls still work.

### Testing
1. Open old `Manage Purchases` section url http://calypso.localhost:3000/purchases.
2. Make sure it redirects to http://calypso.localhost:3000/me/purchases.
3. Make sure `Manage Purchases` is highlighted in the sidebar.
4. Go to `Billing History` section.
5. You should see the following url in your browser: http://calypso.localhost:3000/me/purchases/billing.
6. Type http://calypso.localhost:3000/me/billing to make sure it redirects back to the url from (5).
7. Click on `View Receipt` link in one of the receipt rows.
8. You should receipt details.
9. Click `Back` button.
10. You should get back to `Billing History` page.
11. Click `Add Credit Card` button.
12. You should see the following url in your browser: http://calypso.localhost:3000/me/purchases/add-credit-card.
13. Type http://calypso.localhost:3000/payment-methods/add-credit-card to make sure it redirects back to the url from (12).
14. Click `Back` button.
15. You should get back to `Billing History` page.
16. Click `Go to "Purchases" to add or cancel a plan.`.
17. You should see `Manage Purchases` page again.
18. Click on any purchase and then follow navigation links and play around.

All old urls have redirects in place. Pattern is described in `Changes introduced` section.

### Server changes
I'm also working on a backend patch that will update all urls to new patterns.
